addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -56,8 +56,9 @@ class MyModel(models.Model):
                   AND m.res_id IN %(record_ids)s
              ORDER BY v.id
         """
-        self.env.cr.execute(query, {"field_id": field.id, "model_name": self._name, "record_ids": tuple(self.ids)})
-        trackings = self.env.cr.dictfetchall()
+        with self.env.cr.savepoint():
```

Comment:
Can you please explain the role of this in your fix? :)

Reply:
It is used to create a savepoint within the current database transaction. A savepoint is a point in a transaction where we can roll back if needed. 
By using this in our case we will get the values of all the previous stages in `trackings` and when function calls again all the values saved in savepoint will execute. Previously the values in `trackings` were lost when function calls again.

So to not keep old value in the next stage in statusbar i have implied this.

Reply:
*Small summary*

So here is what I understand of the situation, and how it is leading to the issue. The compute and `_get_duration_from_tracking` method work as expected. However, it computes the durations based on existing records of mail.tracking.value. (+1 fake for current stage)

The issue is that when, on the form view, one changes the stage, the _compute will NOT fetch the lastly created mail.tracking.value (the one tracking the stage change) in the dictfetchall. The weird duration jump from one stage to the other comes from there.

However, if you leave the form view, and go back to it, it will be correctly handled. So the issue is that the _compute would need to have updated values when staying on the same form view.

The savepoint does the trick but feels arbitrary, as what we need is not previous values, but all values including the last one, actually. I would have thought flushes would be enough. I'm still thinking about it.

Reply:
Hello @Odoonan,

Flushes seems to be working, so should I replace savepoint with flush. Kindly give your suggestions.
Thanks!!

Reply:
Hello @sauc-odoo , yes you can update so that I see what you did :)

We can always go back to this later.

Cheers!

Reply:
Updated the code as per the suggestions, kindly check it.
Thanks!!